index.ts

```diff
@@ -68,8 +68,7 @@ async function get<TValue extends Value>(key: string): Promise<TValue | undefine
 
 async function set<TValue extends Value>(key: string, value: TValue, maxAge: TimeDescriptor = {days: 30}): Promise<TValue> {
 	if (typeof value === 'undefined') {
-		// @ts-ignore This never happens in TS because `value` can't be undefined
-		return;
+		return delete_(key);
```

Comment:
To solve the type issue, you can try:

```suggestion
		await delete_(key);
		return value;
```